### DISCUSSION TITLE
Migrations with foreign keys

### DISCUSSION THREAD

**Discussion by Style87** [phalcon/cphalcon#16185] [Q&A] [answered]
I made a new app and an accompanying mysql database. I have a grand total of 6 tables some of which contain foreign keys to the others. I executed `vendor/bin/phalcon-migrations generate` to generate an initial migration. I deployed to my production environment and ran `vendor/bin/phalcon-migrations run`. I got the error `Failed to open the referenced table...`. It seems as though the migration code doesn't understand the schema structure and the order in which to execute the migration files. Some quick googling lead me to a [github post](https://github.com/phalcon/phalcon-devtools/issues/556) from 2015 that was closed in 2017 with no satisfactory resolution. As some five years have passed, is there now a way to seamlessly resolve this using the migration tools? Right now my options seem to be to create all tables without foreign keys, generate migrations, add foreign keys, and generate migrations again or create all independent tables, generate migrations, create tables that reference the independent tables, and generate migrations again. Either way is a process based workaround for a bug that should be resolved in the tool.

**Comment by Jeckerson** [comment, marked as answer]
Hello, are you using latest version of `phalcon/migrations` (https://github.com/phalcon/migrations) package? Also, if you are using via devtools, please use separately via composer `composer require --dev phalcon/migrations`.

If your problem persists, please open new issue here https://github.com/phalcon/migrations/issues .

**Reply by Style87** [reply]
Thank you for the response. My issue continues to persist after removing phalcon/devtools and adding phalcon/migrations. I'll open an issue in the proper repo.
